addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/ExternalRedirectScanRule.java

```diff
@@ -478,10 +485,42 @@ private static RedirectType isRedirected(String payload, HttpMessage msg) {
 
     private static boolean isRedirectPresent(Pattern pattern, String value) {
         Matcher matcher = pattern.matcher(value);
+        if (!isPresent(matcher)) {
+            return false;
+        }
+        Set<String> extractedComments = extractJsComments(value);
+        String valueWithoutComments = value;
+        for (String comment : extractedComments) {
+            valueWithoutComments = valueWithoutComments.replace(comment, "");
+        }
+
+        return isPresent(pattern.matcher(valueWithoutComments));
+    }
+
+    private static boolean isPresent(Matcher matcher) {
         return matcher.find()
                 && StringUtils.startsWithIgnoreCase(matcher.group(1), HttpHeader.HTTP);
     }
 
+    /** Visibility increased for unit testing purposes only */
+    protected static Set<String> extractJsComments(String jsSource) {
+        Set<String> comments = new HashSet<>();
+        try {
+            CompilerEnvirons env = new CompilerEnvirons();
+            env.setRecordingComments(true);
+            Parser parser = new Parser(env, env.getErrorReporter());
+            // Rhino drops a character when the snippet ends with a single line comment so add a newline
+            AstRoot ast = parser.parse(jsSource + "\n", null, 1);
```

Comment:
This can be seen by leveraging `debugPrint()` on the ast or comment, the length of the comment is always reported one character short 😞

Reply:
Reported upstream: https://github.com/mozilla/rhino/issues/2151

Reply:
Without a doubt when others read this on Monday you'll wonder why this wasn't relevant before. But you can see from the simple example in the linked issue that it is a Rhino issue. I'm not sure, I didn't get to dig into that yet. 

I can but I don't want it to derail this. Let me know if you prefer that I roll back to the previous state.